app/services/project_life_cycle_steps/reschedule_service.rb

```diff
@@ -48,16 +48,24 @@ def persist(service_call)
     def reschedule_phases(phases:, from:)
       phases.each do |phase|
         next unless phase.active?
-        next unless phase.date_range_set?
-        next unless phase.duration&.positive? # not updated using service that sets duration
 
-        date_range = calculate_date_range(from, duration: phase.duration)
-        next unless date_range
+        if phase.date_range_set?
+          next_start_date = set_phase_dates_and_next_start(phase, from)
+          from = next_start_date unless next_start_date.nil?
```

Comment:
```suggestion
          from = set_phase_dates_and_next_start(phase, from) || from
```

Reply:
I believe it reads a bit better to have the condition instead of having just the `||`.